packages/ckeditor5-ui/src/focuscycler.ts

```diff
@@ -302,11 +339,40 @@ export interface FocusCyclerActions {
 	focusPrevious?: ArrayOrItem<string>;
 }
 
+/**
+ * Fired when the focus cycler is about to move the focus from the last focusable item
+ * to the first one.
+ *
+ * @eventName ~FocusCycler#forwardCycle
+ */
+export type FocusCyclerForwardCycleEvent = {
+	name: 'forwardCycle';
+	args: [];
+};
+
+/**
+ * Fired when the focus cycler is about to move the focus from the first focusable item
+ * to the last one.
+ *
+ * @eventName ~FocusCycler#backwardCycle
+ */
+export type FocusCyclerBackwardCycleEvent = {
+	name: 'backwardCycle';
+	args: [];
+};
+
 /**
  * Checks whether a view is focusable.
  *
  * @param view A view to be checked.
  */
-function isFocusable( view: View & { focus?: unknown } ) {
-	return !!( view.focus && isVisible( view.element ) );
+function isPlainFocusableView( view: View ): view is FocusableView {
```

Comment:
What is a "plain focusable view"? AFAICS, the implementation did not change. If this is to differentiate "focusable" from "focusable with focusable children", I wouldn't make a change. To me, it felt like maybe "plain focsuable view" is a focusable that is not a "focusable with focusable children". But that's not true. I might not get something, but at the first glance, the change and the new name is confusing.